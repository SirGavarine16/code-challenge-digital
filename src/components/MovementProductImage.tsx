import React, { FC, useEffect } from 'react';
import { Animated, Image, StyleSheet, View } from 'react-native';

import { colors } from '../themes/AppTheme';
import { useAnimations } from '../hooks';

interface Props {
    image: string;
}

const MovementProductImage: FC<Props> = ({ image }) => {
    const { container, content } = styles;

    const { fadeIn, fadeOut, movePosition, opacity, position } = useAnimations();

    useEffect(() => {
        fadeIn();
        movePosition(-100, 500);
        return () => {
            fadeOut();
        }
    }, []);

    return (
        <Animated.View testID='card-image' style={[container, { opacity, transform: [{ translateX: position }] }]}>
            <Image testID='product-image' source={{ uri: image }}  style={content} resizeMode='contain' />
        </Animated.View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        width: '100%',
        aspectRatio: 1 / 1,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 4,
        },
        shadowOpacity: 0.30,
        shadowRadius: 4.65,
        elevation: 8,
        borderRadius: 20,
        justifyContent: 'center',
        alignItems: 'center',
    },
    content: {
        width: '100%',
        height: '100%',
        backgroundColor: colors.background
    }
})

export default MovementProductImage;
