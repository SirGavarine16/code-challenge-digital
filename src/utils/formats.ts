import dayjs from 'dayjs';
import 'dayjs/locale/es-mx';

export const formatToScore = (number: number) => {
    const score = new Intl.NumberFormat('es-MX', { style: 'currency', currency: 'MXN' }).format(number).replaceAll('$', '');
    return `${score} pts`;
}

export const formatDate = (date: string) => {
    return dayjs(date).locale('es-mx').format('D [de] MMMM, YYYY');
}