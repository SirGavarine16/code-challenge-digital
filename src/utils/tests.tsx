import { PropsWithChildren } from 'react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import { rootReducer } from '../redux/store';

export const SafeAreaWrapper = ({ children }: PropsWithChildren<{}>) => {
    return <SafeAreaProvider>{children}</SafeAreaProvider>
}

export const ReduxWrapper = ({ children }: PropsWithChildren<{}>) => {
    const store = configureStore({ reducer: rootReducer });
    return <Provider store={store}>{children}</Provider>
}