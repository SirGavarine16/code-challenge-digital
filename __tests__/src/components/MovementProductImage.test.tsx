import { render } from '@testing-library/react-native';

import MovementProductImage from '../../../src/components/MovementProductImage';
import { dummyData } from '../../../src/utils/tests';

describe('<MovementProductImage />' , () => {
    const renderComponent = () => render(<MovementProductImage image={dummyData.image} />);

    it('should render correctly', () => {
        expect(renderComponent()).toMatchSnapshot();
    });

    it('should render an image card', () => {
        const { getByTestId } = renderComponent();
        expect(getByTestId('card-image')).toBeDefined();
    });

    it('should render an image', () => {
        const { getByTestId } = renderComponent();
        expect(getByTestId('product-image')).toBeDefined();
    });
});