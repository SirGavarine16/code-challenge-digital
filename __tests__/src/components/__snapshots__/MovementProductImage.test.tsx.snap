// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovementProductImage /> should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "aspectRatio": 1,
      "backgroundColor": "white",
      "borderRadius": 20,
      "elevation": 8,
      "justifyContent": "center",
      "opacity": 0,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4.65,
      "transform": [
        {
          "translateX": -100,
        },
      ],
      "width": "100%",
    }
  }
  testID="card-image"
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://loremflickr.com/640/480/technics",
      }
    }
    style={
      {
        "backgroundColor": "#F8F8F8",
        "height": "100%",
        "width": "100%",
      }
    }
    testID="product-image"
  />
</View>
`;
